bmcs/src/main/scala/akka/stream/alpakka/oracle/bmcs/BmcsSettings.scala

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.alpakka.oracle.bmcs
+
+import java.nio.file.{Path, Paths}
+
+import akka.actor.ActorSystem
+import com.typesafe.config.Config
+
+final case class Proxy(host: String, port: Int, scheme: String)
+
+final case class BmcsSettings(bufferType: BufferType, proxy: Option[Proxy], region: String, namespace: String) {
+
+  override def toString: String =
+    s"BmcsSettings($bufferType,$proxy,$region,$namespace)"
+}
+
+sealed trait BufferType {
+  def path: Option[Path]
+}
+
+case object MemoryBufferType extends BufferType {
+  def getInstance: BufferType = MemoryBufferType
+
+  override def path: Option[Path] = None
+}
+
+case class DiskBufferType(filePath: Path) extends BufferType {
+  override val path: Option[Path] = Some(filePath).filterNot(_.toString.isEmpty)
+}
+
+object DiskBufferType {
+  def create(filePath: Path): DiskBufferType = DiskBufferType(filePath)
+}
+
+object BmcsSettings {
+
+  def apply(region: String, namespace: String, bufferType: BufferType = MemoryBufferType)(
+      implicit config: Config
+  ): BmcsSettings =
+    apply(config).copy(region = region, namespace = namespace, bufferType = bufferType)
+
+  /**
+   * Scala API: Creates [[BmcsSettings]] from the [[com.typesafe.config.Config]] attached to an [[akka.actor.ActorSystem]].
+   */
+  def apply()(implicit system: ActorSystem): BmcsSettings = apply(system.settings.config)
+
+  def apply(config: Config): BmcsSettings = {
+
+    val bufferType = config.getString("akka.stream.alpakka.bmcs.buffer") match {
```

Comment:
this is the path that should not be full, it should look for `buffer`, `disk-buffer-path` etc. here  and extract the full path in `apply()(system)`. 

That makes it possible to have a custom config that does not have to live under the full path that you pass to this apply while still using the full path as default when you do not pass the config but use the implicit actor system factory method above.

Also, the config should not be inside the `akka` namespace as this is not Akka core (even if you saw some other connectors may be doing this). I think a good scope for the config would be `alpakka.bmcs`